### PR TITLE
8272392 Lanai: SwingSet2. Black background on expanding tree node

### PR DESCRIPTION
**Pull request by avu**
Removed creation of the separate encoder depending on destination properties as we don't use this info to customize the encoder properties

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8272392](https://bugs.openjdk.java.net/browse/JDK-8272392): Lanai: SwingSet2. Black background on expanding tree node

### Reviewers
 * [Ajit Ghaisas](https://openjdk.java.net/census#aghaisas) (@aghaisas - Committer)
 * [Jayathirth D V](https://openjdk.java.net/census#jdv) (@jayathirthrao - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/6529/head:pull/6529` \
`$ git checkout pull/6529`

Update a local copy of the PR: \
`$ git checkout pull/6529` \
`$ git pull https://git.openjdk.java.net/jdk pull/6529/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 6529`

View PR using the GUI difftool: \
`$ git pr show -t 6529`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/6529.diff">https://git.openjdk.java.net/jdk/pull/6529.diff</a>

</details>

### CODE REVIEW

**Comment by victordyakov** [comment]
@jayathirthrao @aghaisas please review

File: src/java.desktop/macosx/native/libawt_lwawt/java2d/metal/EncoderManager.m
Comment (jayathirthrao) [inline src/java.desktop/macosx/native/libawt_lwawt/java2d/metal/EncoderManager.m:44]: I think by mistake isText argument is also removed. It is not matching reset() implementation.

File: src/java.desktop/macosx/native/libawt_lwawt/java2d/metal/MTLClip.m
Comment (jayathirthrao) [inline src/java.desktop/macosx/native/libawt_lwawt/java2d/metal/MTLClip.m:61]: This looks like cleanup and not related to this change. Sometime back i also had question on why this was needed. Thanks.

**Comment by jayathirthrao** [comment]
I used test case present in https://bugs.openjdk.java.net/browse/JDK-8267963 to check how it behaves with this PR and noticed that it is causes regresssion related to how ruler text looks and also if we move Ruler between monitors it doesnt draw.

Thought this may be because of state's not getting reset since there is mismatch in EncoderManager.reset() arguments. Updated code to add isText argument over this PR, but i continue to see the regressions.

**Comment by jayathirthrao** [comment]
jtreg & JCK tests are not showing any major failures in our CI with this patch.

File: src/java.desktop/macosx/native/libawt_lwawt/java2d/metal/EncoderManager.m
Comment (avu) [reply src/java.desktop/macosx/native/libawt_lwawt/java2d/metal/EncoderManager.m:44]: Thanks for the findings! Fixed.

**Comment by avu** [comment]
> I used test case present in https://bugs.openjdk.java.net/browse/JDK-8267963 to check how it behaves with this PR and noticed that it is causes regresssion related to how ruler text looks and also if we move Ruler between monitors it doesnt draw.
> 
> Thought this may be because of state's not getting reset since there is mismatch in EncoderManager.reset() arguments. Updated code to add isText argument over this PR, but i continue to see the regressions.

@jayathirthrao  could you please provide some more info about your environment. I cannot reproduce the problems that you've reported with the test from JDK-8267963.

**Comment by jayathirthrao** [comment]
> > I used test case present in https://bugs.openjdk.java.net/browse/JDK-8267963 to check how it behaves with this PR and noticed that it is causes regresssion related to how ruler text looks and also if we move Ruler between monitors it doesnt draw.
> > Thought this may be because of state's not getting reset since there is mismatch in EncoderManager.reset() arguments. Updated code to add isText argument over this PR, but i continue to see the regressions.
> 
> @jayathirthrao could you please provide some more info about your environment. I cannot reproduce the problems that you've reported with the test from JDK-8267963.

Strange that i am not seeing the issue i noticed last week in my system. SwingSet2, J2DDemo and Font2DTest are not showing any regressions. I am running some tests with Metal API validation. If tests are green i will approve. @aghaisas also wants to run some tests on discrete graphics, so lets wait for his approval also.

**Review by aghaisas — APPROVED** [review]
I tested functionality using SiwngSet2, J2DDemo and NetBeans IDE on discrete graphics card. I also tested switching to integrated graphics card. I did not observe any regression.

The code changes look good.

**Review by jayathirthrao — APPROVED** [review]

**Pull request closed without merging**
